primitives/src/lib.rs

```diff
@@ -100,7 +100,7 @@ pub enum CurrencyId {
 
 pub const TOKEN_DECIMAL: u128 = 1_000_000_000_000_000_000;
 
-pub const RATE_DECIMAL: u128 = 1_000_000_000;
+pub const RATE_DECIMAL: u128 = 1_000_000_000_000_000_000;
```

Comment:
I think we need to take more concerns here, for type u128: 2^128≈10^38,
suppose when we calculate with RATE_DECIMAL&TOKEN_DECIMAL,  
10^18 * 10^18 * 10^18 ≈ 10^54 will overflow here. we need special care of this...

Reply:
Thank you for your good attention. That is my next refactor work; I will use [Substrate-fixed](https://substrate.dev/recipes/fixed-point.html#substrate-fixed-accumulator) which will build in the floating-point instead of the manual calculating with `RATE_DECIMAL`.